### PR TITLE
FISH-5639 Added warning message to change-master-password command

### PR DESCRIPTION
**Pull request by kalinchan**
## Description

Added warning message to change-master-password command to warn about additional trust and key stores not being re-encrypted 
Message when there are no additional trust and keystores
Message when there are additional stores

warningmessage2

### Testing Performed

Ran command to see if warning message appeared.
### Testing Environment

openjdk version "1.8.0_292" 
ubuntu 20.04 
maven 3.6.3

### CODE REVIEW

**Comment by kalinchan** [comment]
jenkins test

File: nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ChangeMasterPasswordCommand.java [hunk 1]
```diff
@@ -136,13 +136,15 @@ protected int executeCommand() throws CommandException {
...
+                
                 command = CLICommand.getCommand(habitat,
-                        CHANGE_MASTER_PASSWORD_NODE);
+                CHANGE_MASTER_PASSWORD_NODE);
                 return command.execute(newargs);
             }
         } catch (IOException e) {
             throw new CommandException(e.getMessage(),e);
+        } finally {
+            System.out.println("WARNING: Additional Truststores are not re-encrypted, it is your responsibility to do this.");
```
Comment (Pandrex247) [inline nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ChangeMasterPasswordCommand.java:147]: I wouldn't use System.out - its use is generally discouraged.

Use `Logger.getLogger(CLICommand.class.getPackage().getName());`

**Review by Pandrex247 — CHANGES_REQUESTED** [review]
I think this is too simple - it'll print out the warning even if you haven't got any additional key or trust stores. It should print the warning if they've been configured and not otherwise to prevent confusion.

I'd also change the message to be a bit less technical, but also make it more specific by pointing at the actual stores - something along the lines of:
 `The passwords of additional key stores xyz,asd,ijk and trust stores lkj,gfd,cvb have not been changed - please update these manually to continue using them.`

**Comment by kalinchan** [comment]
jenkins test

File: nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java [hunk 2]
```diff
@@ -659,4 +656,37 @@ protected String getInstallRootPath() throws CommandException {
...
+            DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+            DocumentBuilder documentBuilder = documentBuilderFactory.newDocumentBuilder();
+            Document document = documentBuilder.parse(getDomainXml());
+            NodeList jvmOptionsNodes = document.getElementsByTagName("jvm-options");
+
+            for (int i = 0; i < jvmOptionsNodes.getLength(); i++) {
+                if (additionalTrustandKeyStores.containsKey("additionalKeyStores")
+                        && additionalTrustandKeyStores.containsKey("additionalTrustStores")) {
+                    break;
+                }
```
Comment (Pandrex247) [inline nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java:672]: I don't think you should exit the loop here.

Additional key & trust stores are defined at the config level, which typically goes like this:
1. server-config
2. default-config
3. extra-config1
4. extra-config2
5. ...

If you defined extra stores in both the server-config (for the DAS) and in extra-config1 (for an instance), you would only be warned about needing to change the stores for those defined in server-config.

Also this has a small flaw in that if you only defined additional key stores, your results would get overwritten. For example, if you defined additional key stores in server-config and default-config, only the additional keystores in default-config would be listed.

So I think you should drop this exit clause and iterate over the entire list, and make sure you append to the value if there's one present (see [Map.computeIfPresent](https://docs.oracle.com/javase/8/docs/api/java/util/Map.html#computeIfPresent-K-java.util.function.BiFunction-))

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

File: nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java [hunk 3]
```diff
@@ -659,4 +660,50 @@ protected String getInstallRootPath() throws CommandException {
...
+                    } else {
+                        additionalTrustandKeyStores.computeIfPresent("additionalTrustStores", (key, value) -> value.concat(", " + additionalTrustStores));
+                    }
+                    continue;
+                }
+            }
+            if (additionalTrustandKeyStores.containsKey("additionalKeyStores")) {
+                logger.log(Level.INFO,
+                        "The passwords of additional KeyStores {0} have not been changed - please update these manually to continue using them.",
+                        Arrays.toString(additionalTrustandKeyStores.get("additionalKeyStores").split(":")));
```
[the remark this answers is not in the document]
Comment (Pandrex247) [reply nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java:696]: Not strictly related to this PR, but I suspect the additional key/trust store functionality doesn't work on Windows for the same reason as this...

**Review by Pandrex247 — CHANGES_REQUESTED** [review]

**Comment by kalinchan** [comment]
jenkins test

**Comment by kalinchan** [comment]
jenkins test

File: nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java [hunk 4]
```diff
@@ -659,4 +660,41 @@ protected String getInstallRootPath() throws CommandException {
         }
         return installRootPath;
     }
-}
+
+    protected void checkAdditionalTrustandKeyStores() throws IOException, XMLStreamException {
```
Comment (Pandrex247) [inline nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/LocalServerCommand.java:664]:
```suggestion
    protected void checkAdditionalTrustAndKeyStores() throws IOException, XMLStreamException {
```

File: nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ChangeNodeMasterPasswordCommand.java [hunk 5]
```diff
@@ -133,13 +135,19 @@ protected void validate() throws CommandException, CommandValidationException {
...
             }
         } catch (IOException | XMLStreamException exception) {
-            LOGGER.warning("Could not determine if data grid encryption is enabled - " +
-                    "you will need to regenerate the encryption key if it is");
+            LOGGER.warning("Could not determine if data grid encryption is enabled - "
+                    + "you will need to regenerate the encryption key if it is");
         }
+
+        try {
+            checkAdditionalTrustandKeyStores();
```
Comment (Pandrex247) [inline nucleus/cluster/cli/src/main/java/com/sun/enterprise/admin/cli/cluster/ChangeNodeMasterPasswordCommand.java:146]:
```suggestion
            checkAdditionalTrustAndKeyStores();
```

File: nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ChangeMasterPasswordCommandDAS.java [hunk 6]
```diff
@@ -118,44 +105,47 @@ protected int executeCommand() throws CommandException {
...
                 }
             } catch (IOException | XMLStreamException exception) {
-                logger.warning("Could not determine if data grid encryption is enabled - " +
-                        "you will need to regenerate the encryption key if it is");
+                logger.warning("Could not determine if data grid encryption is enabled - "
+                        + "you will need to regenerate the encryption key if it is");
+            }
+
+            try {
+                checkAdditionalTrustandKeyStores();
```
Comment (Pandrex247) [inline nucleus/admin/server-mgmt/src/main/java/com/sun/enterprise/admin/servermgmt/cli/ChangeMasterPasswordCommandDAS.java:140]:
```suggestion
                checkAdditionalTrustAndKeyStores();
```

**Review by Pandrex247 — APPROVED** [review]
One super minor change, but otherwise 👍

**Comment by kalinchan** [comment]
jenkins test

**Review by aubi — COMMENTED** [review]
--deleted, mistake--

**Pull request merged**